yajsapi/executor/index.ts

```diff
@@ -600,26 +604,46 @@ export class Executor {
           );
         }
         const task_id = current_worker_task ? current_worker_task.id : undefined;
-        batch.attestation = {
-          credentials: activity.credentials,
-          nonce: activity.id,
-          exeunitHashes: activity.exeunitHashes,
-        };
+        // batch.attestation = {
+        //   credentials: activity.credentials,
+        //   nonce: activity.id,
+        //   exeunitHashes: activity.exeunitHashes,
+        // };
         await batch.prepare();
         const cc = new CommandContainer();
         batch.register(cc);
-        const remote = await activity.send(
-          cc.commands(),
-          stream_output,
-          batch_deadline,
-          (<SubmissionState>self.state).worker_cancellation_token
-        );
         const cmds = cc.commands();
+
+        const script = new Script(cmds.map((cmd) => new Command(Object.keys(cmd)[0], cmd[Object.keys(cmd)[0]])));
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const cancellationToken = (<SubmissionState>self.state).worker_cancellation_token as CancellationToken;
+
+        const scriptResults = await activity
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .execute(script.getExeScriptRequest(), false, batch_deadline, cancellationToken)
+          .catch((e) => {
+            throw new CommandExecutionError("", e.toString());
+          });
+
         emit(new events.ScriptSent({ agr_id: agreement_id, task_id, cmds }));
+
         const get_batch_results = async (): Promise<events.CommandEvent[]> => {
           const results: events.CommandEvent[] = [];
-          for await (const evt_ctx of remote) {
-            const evt = evt_ctx.event(agreement_id, task_id, cmds);
+          scriptResults.on("error", (error) => {
+            throw new CommandExecutionError("", error.toString());
+          });
+          for await (const result of scriptResults) {
+            let evt = Object.create(events.CommandExecuted.prototype);
```

Comment:
please extract the creation of this event somewhere (either to the workcontext or to the activity module) instead of coding it inline here...